### ISSUE TITLE
✨ Train-time ONNX export logs model to active W&B run

### ISSUE THREAD

**Issue by CedrickArmel** [CedrickArmel/radiologist#109]
### Context

Implements the opt-in end-of-fit export behavior and the uploader **log** half of the split. This slice
replaces the skeleton stubs for `OnnxExportCallback.on_fit_end` and
`_WandbUploader.log_model_artifacts`, reaching GREEN-real (no `NotImplementedError` reachable through
the callback). It reuses `export_onnx()` verbatim and the plain-`wandb` active-run convention from
`WandbDefineSummaryCallback`. See the epic milestone for the artifact-layout rationale. Requires: #108.

### User story

As a **model trainer**, I want a finished training run to automatically export ONNX and log it to my
active W&B run so that **the model is preserved and later promotable without a local re-pull** — but
only when I explicitly opt in.

### Acceptance criteria

- [ ] Given a completed `trainer.fit` with an active W&B run and a written best checkpoint, the active
      run receives a logged `model-{run_id}` artifact carrying alias `best` and a `model-{run_id}-mcd`
      artifact carrying alias `best`.
- [ ] Given a completed fit that also produced a last checkpoint, a `model-{run_id}` version carrying
      alias `last` is also logged.
- [ ] After the callback runs, the deterministic and MC-Dropout `.onnx` files exist on disk in the run
      output directory.
- [ ] When no W&B run is active, the callback makes no W&B calls and writes nothing (silent no-op).
- [ ] When no best checkpoint path is available, the callback makes no W&B calls (silent no-op).
- [ ] Given a logged-but-not-linked artifact, no collection link / `production` / `staging` alias is
      applied by this slice.
- [ ] mypy clean; pytest green.

### Data flow

```
on_fit_end → wandb.run active? → best_model_path from trainer.checkpoint_callback
           → export_onnx(...) → det.onnx + mcd.onnx on disk
           → WandbRegistry.log_model_artifacts(result, run, best_ckpt, last_ckpt)
           → run.log_artifact(model-{run_id} [best.ckpt+det.onnx], aliases=['best'])
           → run.log_artifact(model-{run_id}-mcd [mcd.onnx], aliases=['best'])
           → run.log_artifact(model-{run_id} [last.ckpt], aliases=['last'])   # if last exists
```

### Technical notes

- `radiologist-core/src/radiologist/core/registry/export.py:97-189` — `export_onnx(ckpt_path, run_id,
  input_shape, classes, cam_target_layer, out_dir, opset=18) -> ExportResult` already exists and works;
  call it unchanged. `out_dir = trainer.log_dir or trainer.default_root_dir`.
- `radiologist-core/src/radiologist/core/callbacks/wandb_summary.py` — copy its
  `getattr(wandb, "run", None)` no-op guard exactly; the new callback owns its own `import wandb`, not
  the registry `_wandb` sentinel.
- `radiologist-core/src/radiologist/core/configs/callbacks/model_checkpoint.yaml` configures Lightning's
  `ModelCheckpoint` (`save_top_k: 1`, `save_last: true`, `monitor: val_score`, `mode: max`); the best
  checkpoint source is `trainer.checkpoint_callback.best_model_path` (same accessor already used in
  `train.py:119` for the test stage); the last checkpoint source is
  `trainer.checkpoint_callback.last_model_path`.
- `_WandbUploader.log_model_artifacts` adds files via `art.add_file(path)`; the det artifact gets
  **both** `export_result.det_path` and `ckpt_path`. Use `run.log_artifact(art, aliases=[...])` — do
  **not** call `wandb.init` (the active run is injected) and do **not** call `run.link_artifact`
  (that belongs to the separate `link_to_collection` seam, not this slice).
- Test pattern (classist outside-in TDD — never mock owned code, only the W&B SDK boundary): real
  `lightning.pytorch.Trainer` with a real `ModelCheckpoint` (small `limit_train_batches`/`max_epochs=1`,
  **not** `fast_dev_run`, so `best_model_path` is populated) + real `lmodule`/`dm` fixtures from
  `radiologist-core/tests/conftest.py`; only the W&B run is faked (`patch.object(wandb, "run",
  MagicMock())`); assert `fake_run.log_artifact` call args (alias lists) and assert the onnx files exist
  on disk. The no-op tests use the existing `fast_dev_run=True` pattern (mirror
  `radiologist-core/tests/test_callbacks.py`). Do not mock `export_onnx` — it is owned code and must run
  for real.
